### ISSUE TITLE
catch2 sigStackSize errors

### ISSUE THREAD

**Issue by maedoc** [celerity/celerity-runtime#87]
Building against hipSYCL, CUDA 11.6, clang-12, Ubuntu 21.10, the build errors on 
```
[ 63%] Built target convolution
[ 65%] Building CXX object test/CMakeFiles/unit_test_suite.dir/unit_test_suite.cc.o
clang: warning: Unknown CUDA version. cuda.h: CUDA_VERSION=11060. Assuming the latest supported version 10.1 [-Wunknown-cuda-version]
In file included from /home/duke/src/celerity-runtime/test/unit_test_suite.cc:4:
/home/duke/src/celerity-runtime/vendor/Catch2/single_include/catch2/catch.hpp:10675:34: error: constexpr variable 'sigStackSize' must be initialized by a constant expression
    static constexpr std::size_t sigStackSize = 32768 >= MINSIGSTKSZ ? 32768 : MINSIGSTKSZ;
                                 ^              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/duke/src/celerity-runtime/vendor/Catch2/single_include/catch2/catch.hpp:10675:58: note: non-constexpr function 'sysconf' cannot be used in a constant expression
    static constexpr std::size_t sigStackSize = 32768 >= MINSIGSTKSZ ? 32768 : MINSIGSTKSZ;
                                                         ^
/usr/include/x86_64-linux-gnu/bits/sigstksz.h:32:22: note: expanded from macro 'MINSIGSTKSZ'
# define MINSIGSTKSZ SIGSTKSZ
                     ^
/usr/include/x86_64-linux-gnu/bits/sigstksz.h:28:19: note: expanded from macro 'SIGSTKSZ'
# define SIGSTKSZ sysconf (_SC_SIGSTKSZ)
                  ^
/usr/include/unistd.h:640:17: note: declared here
extern long int sysconf (int __name) __THROW;
                ^
In file included from /home/duke/src/celerity-runtime/test/unit_test_suite.cc:4:
/home/duke/src/celerity-runtime/vendor/Catch2/single_include/catch2/catch.hpp:10675:34: error: dynamic initialization is not supported for __device__, __constant__, __shared__, and __managed__ variables.
    static constexpr std::size_t sigStackSize = 32768 >= MINSIGSTKSZ ? 32768 : MINSIGSTKSZ;
                                 ^              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 errors generated when compiling for sm_75.
```
Could it be an issue of the unknown CUDA version defaulting to 10.1?

(unrelated: I'm looking at your CI script to figure out how get a working setup, but I haven't figured them out yet.  Where/how are the containers built?)

**Comment by maedoc** [comment]
perhaps helpful, the configure results
```
~/src/celerity-runtime/build$ cmake .. -DHIPSYCL_TARGETS=cuda:sm_75 -DCMAKE_C_COMPILER=clang-12 -DCMAKE_CXX_COMPILER=clang++-12
-- The CXX compiler identification is Clang 12.0.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++-12 - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found MPI_CXX: /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_cxx.so (found suitable version "3.1", minimum required is "2.0") 
-- Found MPI: TRUE (found suitable version "3.1", minimum required is "2.0")  
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found hipSYCL: /usr/local/lib/cmake/hipSYCL
-- Looking for CL_VERSION_2_2
-- Looking for CL_VERSION_2_2 - found
-- Found OpenCL: /usr/local/cuda/lib64/libOpenCL.so (found version "2.2") 
-- Automatically chooosing hipSYCL because it is the only SYCL implementation available
-- hipSYCL does NOT support SYCL 2020 reductions
-- Build spdlog: 1.4.0
-- Build type: Release
-- Checking for one of the modules 'hdf5-openmpi;hdf5-1.12.0;hdf5'
-- Configuring done
-- Generating done

```

**Comment by PeterTh** [comment]
Hi, thanks for the report!

You seem to be running into an upstream bug in catch2:
https://github.com/catchorg/Catch2/issues/2178

That was fixed a while ago, probably all that is needed is updating our catch2 reference (which is way outdated).

**Comment by PeterTh** [comment]
Please check this again with the latest version and re-open this issue if it is not resolved.

**Comment by PeterTh** [comment]
@maedoc, I forgot about your question regarding the CI while I focused on this error. Have you seen the celerity-runtime-ci repo? https://github.com/celerity/celerity-runtime-ci

**Comment by psalz** [comment]
> @maedoc, I forgot about your question regarding the CI while I focused on this error. Have you seen the celerity-runtime-ci repo? [celerity/celerity-runtime-ci](https://github.com/celerity/celerity-runtime-ci)

That repo is private I'm afraid. With this bug fixed, is there anything else blocking your setup?

**Comment by maedoc** [comment]
No, I think I've got it figured out well enough for now, and I need to get busy learning SYCL & Celerity.   I had to iterate several times to figure out that Ubuntu 20.04 or 21.10 with Clang 12 & OpenMPI and hipSYCL (and then Celerity as well) seems to be the way to go right now, and access to a CI set up would be helpful to see how you're doing it.  You might consider a Dockerfile or Singularityfile just for documentation purposes, for example [here's mine for hipSYCL](https://github.com/maedoc/hipSYCL/blob/develop/Dockerfile) and Celerity is just a git/cmake/make on top of that.
